### ISSUE TITLE
"Use another account"

### ISSUE THREAD

**Issue by GithubAshishHere** [AzureAD/microsoft-authentication-library-for-objc#850]
Not working: Login successfully with one user and then without killing the application and again try to login "Use another account". It will ask for the new email to enter. After entering a new email and click on next. it's automatically getting old signed in account and login with that old account successfully.

Working: if I login successfully with one account and kill application then again open the application and go to the option "Use another account". Enter a new email address now it's asking for the password and login with new enter email.

In short why I need to kill application to log in with Use another account.
How can I login with another account without kill the application?
Waiting for the response. Thank you.

**Comment by oldalton** [comment]
Looks similar to this issue: https://github.com/AzureAD/microsoft-authentication-library-for-objc/issues/560. Can you try suggestion in the end of the issue discussion?

> Can you try setting promptType as selectAccount?
> 
> https://azuread.github.io/microsoft-authentication-library-for-objc/Classes/MSALInteractiveTokenParameters.html#/c:objc(cs)MSALInteractiveTokenParameters(py)promptType

Are your users from the same domain? e.g. user1@contoso.com and user2@contoso.com. If yes, than you might be hitting ADFS behavior where multiple users are not supported, and you need to pass promptType as Login to override it.

**Comment by GithubAshishHere** [comment]
Yes, I'm passing like this way still having the issue.
   MSALInteractiveTokenParameters *interactiveParams = [[MSALInteractiveTokenParameters alloc] initWithScopes:scopes webviewParameters:webParameters];
 [interactiveParams setPromptType:MSALPromptTypeSelectAccount];

Can you please follow the process. Your application is fresh. WebView Open to login with the new user account ex: abc@xyx.com after that success with the first email. Now I'm getting an option Already signed in (abc@xyz.com) and another option is "Use another account". when I click on use another account option. I'm getting a new textbox to enter a new email. after an enter new email and press the next button it redirects with old signed in email id and login successfully with old email id.

**Comment by GithubAshishHere** [comment]
But if before I login to another account. I kill the application then I will able to login with new user account. 
Can you please suggest to me?

**Comment by GithubAshishHere** [comment]
The same thing occurs in SDK demo also.

**Comment by GithubAshishHere** [comment]
Waiting for your response. Thank you

**Comment by jasoncoolmax** [comment]
@GithubAshishHere Seems you are indeed hitting the "ADFS server not supporting multiple users" issue. If ADFS server sees an account in cookies, it will return tokens for that account ignoring your login hint. 
Killing the app helps because it will kill some non persistent cookies in webview. 

Could you provide the correlation id (could be found in MSAL logs)? such that I can further confirm it is an ADFS case.

**Comment by GithubAshishHere** [comment]
@jasoncoolmax Can you please provide me. How can I do login with new login hint without kill application?

**Comment by GithubAshishHere** [comment]
Because same thing working well in android devices.

**Comment by GithubAshishHere** [comment]
Please check below is the correlation id
Correlation Id: 5e4a6db1-7c5c-433e-bf43-aa3cf33de204

**Comment by GithubAshishHere** [comment]
We can't able to clear such a token from the application?

**Comment by jasoncoolmax** [comment]
@GithubAshishHere In your case, if you know you are logging in a new user, you can use [MSALPromptType](https://github.com/AzureAD/microsoft-authentication-library-for-objc/blob/dev/MSAL/src/public/MSALInteractiveTokenParameters.h#L45) to be `MSALPromptTypeLogin` when you call `[MSALPublicClientApplication acquireTokenWithParameters:completionBlock]. In such a way a parameter will be passed down to ADFS server to force user to input credentials and not relying on cookies.

Let me know if that helps solve your issue :)

**Comment by GithubAshishHere** [comment]
Yes, It's good. But every time the user has to enter the password. But I'm talking about behavior `MSALPromptTypeSelectAccount`. It's not working well as I mentioned above. Thank you.

**Comment by jasoncoolmax** [comment]
Unfortunately "multiple accounts signed in" is not supported in ADFS today. That's only workaround that I can come up with. However, you can always try to call acquireTokenSilent before calling acquireToken. The silent call should succeed as long as the refresh token in cache is still valid.

**Comment by GithubAshishHere** [comment]
Okay. Thank you for your important review @jasoncoolmax. I hope in the future. We have the facility to log in. Again thank you.
